project-chip/connectedhomeip
# Add Software Diagnostics Cluster definition

#### Problem

We are missing the Software Diagnostics Cluster definition in CHIP SDK.  

The specification of Software Diagnostics Cluster:
https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/service_device_management/DiagnosticsSoftware.adoc

To help your review, I put all manual added code into the first commit. The rest of commits contain the changes either from the cluster GUI or codegen.

#### Change overview
Add the Software Diagnostics cluster defination based on the CHIP-spec.

#### Testing

This PR only contains the cluster defination part, the implementation part will be done later after Atrribute Read Reques/Response is ready in Interaction Model.

* Run ./out/debug/standalone/chip-tool to get the list of supported clusters and confirm cluster 'softwarediagnostics' is in the list.
* Run ./out/debug/standalone/chip-tool softwarediagnostics to get the list of supported commands for cluster 'softwarediagnostics' and confirm 'reset-watermarks' is   in the list.
